osu.Framework.Android/Platform/AndroidOrientationManager.cs

```
@@ -0,0 +1,67 @@
+
+using Android.Content.PM;
+using osu.Framework.Platform;
+using osu.Framework.Configuration;
+using osu.Framework.Bindables;
+using Android.App;
+
+namespace osu.Framework.Android.Platform
+{
+    public class AndroidOrientationManager : OrientationManager
+    {
+        public ScreenOrientation CurrentScreenOrientation { get; internal set; }
+
+        private AndroidOrientationManager() { }
+        /// <summary>
+        /// Initialize a manager class to control Orientation on Android
+        /// </summary>
+        /// <param name="config">Framework config manager used to get the orientation setting</param>
+        /// <param name="gameActivity">Game activity to initialize orientation control on</param>
+        public AndroidOrientationManager(FrameworkConfigManager config, Activity gameActivity)
+            : base(config)
+        {
+            GameActivity = gameActivity;
+        }
+
+        /// <summary>
+        /// Get/set the game activity that have its orientation controlled.
+        /// </summary>
+        protected Activity GameActivity { get; set; }
+
+        /// <summary>
+        /// Convert <see cref="Orientation"/> framework setting enum to Android's <see cref="ScreenOrientation"/> enum
+        /// </summary>
+        /// <remarks>
+        /// Override this to customize orientation flags
+        /// </remarks>
+        /// <param name="orientation">Orientation</param>
+        /// <returns><see cref="ScreenOrientation"/> enum to use with Android SDK</returns>
+        public virtual ScreenOrientation OrientationToScreenOrientation(Orientation orientation)
+        {
+            switch (orientation)
+            {
+                case Orientation.Landscape:
+                    return ScreenOrientation.UserLandscape;
+                case Orientation.Portrait:
+                    return ScreenOrientation.UserPortrait;
+                case Orientation.Auto:
+                default:
```

I prefer to handle unknown enum members by throwing `ArgumentOutOfRangeException` rather than assuming that `Auto` was wanted as done here.

---

Will be changed in next commits